woocommerce/woocommerce-gutenberg-products-block
# Reduce number of dependencies for the product categories list block

This PR improves how JS powered blocks work on the frontend, reducing the overall number of dependencies. In testing, this reduced downloaded JS from 4.8MB to 2.2MB.

Fixes #762

The changes to the webpack config are to prevent wp.element and lodash being enqueued on the frontend - we can use React only here.

### How to test the changes in this Pull Request:

1. before checking out this branch, create some product category list blocks in a page
2. checkout and build
3. confirm `build/frontend.deps.js` only contains `["react","react-dom","wp-i18n","wp-polyfill"]`
4. check existing list blocks work in the editor and on the frontend
5. create a new product category list block - ensure it works as expected



### Changelog

> Reduced the number of JavaScript assets needed for the Product Category List block to improve performance.
